source/include/VirtualInstrument.h

```diff
@@ -62,12 +62,23 @@ public :
 
 public:
     VirtualInstrument(ExecutionContextRef context, int clumps, TypeRef paramsType, TypeRef localsType);
+    ~VirtualInstrument()               { if (_viName) delete _viName; }
     TypeManagerRef TheTypeManager()     { return _typeManager; }
     TypedObjectRef Params()             { return _params; }
     TypedObjectRef Locals()             { return _locals; }
     TypedArray1D<VIClump>* Clumps()     { return _clumps; }
-    SubString VIName()                  { return _viName; }
-    void SetVIName(SubString s)         { _viName = s; }
+    SubString VIName()                  {
+        SubString s;
+        if (_viName)
+            s.AliasAssignLen(_viName, IntIndex(strlen(ConstCStr(_viName))));
+        return s;
+    }
+    void SetVIName(const SubString *s)   {
```

Comment:
Since SetVIName is not modifying "s", I would make this parameter a const SubString reference instead of pointer to follow the convention. It will also be clear at the call site that the argument being passed is not modified by the caller.